### PR TITLE
[ci] Collect and publish bazel trace

### PR DESCRIPTION
**Pull request by jwnrt**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -244,6 +246,10 @@ jobs:
         --test_tag_filters=-broken,-cw310,-verilator,-dv \
         --target_pattern_file="${TARGET_PATTERN_FILE}"
     displayName: Build & test SW
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathToPublish: '$(Build.ArtifactStagingDirectory)'
+      artifactName: bazel_trace.gz
```
Comment (pamaury) [inline azure-pipelines.yml:252]: I would suggest using a more precise name to specify which bazel trace it is (sw_build).

**Review by pamaury — APPROVED** [review]
Looks good to me and could be quite useful! Thanks James

**Comment by jwnrt** [comment]
The artifact is here if anyone's interested:

[bazel_trace.gz](https://dev.azure.com/lowrisc/_apis/resources/Containers/34927347/bazel_trace.gz?itemPath=bazel_trace.gz%2Fbazel_trace.gz) (warning, will immediately download a double-gzipped JSON file)

Azure seems to have re-gzipped it, so you'll need to `gzip -d bazel_trace.gz` once. Analysis of the traces can be done with <https://ui.perfetto.dev/> or <chrome://tracing> if you use a chromium browser.

I'll see how useful this data is and consider going ahead with this PR. If anyone else finds it useful, let me know.

**Comment by jwnrt** [comment]
I didn't get much insight from these traces and will close this PR, however if anybody else is interested in this then please feel free to re-open.

**Pull request closed without merging**